Packs/qualys/Integrations/QualysEventCollector/QualysEventCollector.py

```diff
@@ -0,0 +1,635 @@
+    """ Removes events with certain time.
+        Args:
+            events: list of events to remove the time from
+            time_to_remove: remove events with this time
+            time_field: the field name where the time is
+    """
+    new_events = []
+    for event in events:
+        if event.get(time_field) == time_to_remove:
+            demisto.debug(f'Removed activity log event with time: {time_to_remove}, log: {event}')
+        else:
+            new_events.append(event)
+    return new_events
+
+
+def add_fields_to_events(events, time_field_path, event_type_field):
+    """
+    Adds the _time key to the events.
+    Args:
+        events: List[Dict] - list of events to add the _time key to.
+        time_field_path: the list of fields to get _time from
+        event_type_field: type field in order to distinguish between the API's
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
+        for event in events:
+            event['_time'] = dict_safe_get(event, time_field_path)
+            event['event_type'] = event_type_field
+
+
+def get_detections_from_hosts(hosts):
+    """
+    Parses detections from hosts.
+    Each host contains list of detections:
+    {'ID':1,
+    'IP': '1.1.1.1',
+    'LAST_VM_SCANNED_DATE': '01-01-2020',
+    'DETECTION_LIST': {'DETECTION': [first_detection_data, second_detection, ...]}
+    'additional_fields': ...
+    }
+
+    The function parses the data in the following way:
+    {''ID':1,
+    'IP': '1.1.1.1',
+    'LAST_VM_SCANNED_DATE': '01-01-2020',
+    'DETECTION': first_detection_data
+    'additional_fields': ...
+    },
+    {'ID':1,
+    'IP': '1.1.1.1',
+    'LAST_VM_SCANNED_DATE': '01-01-2020',
+    'DETECTION': second_detection_data
+    'additional_fields': ...
+    }
+    ....
+
+    :param hosts: list of hosts that contains detections.
+    :return: parsed events.
+    """
+    fetched_events = []
+    for host in hosts:
+        if detections_list := host.get('DETECTION_LIST', {}).get('DETECTION'):
+            if isinstance(detections_list, list):
+                for detection in detections_list:
+                    new_detection = copy.deepcopy(host)
+                    del new_detection['DETECTION_LIST']
+                    new_detection['DETECTION'] = detection
+                    fetched_events.append(new_detection)
+            elif isinstance(detections_list, dict):
+                new_detection = copy.deepcopy(host)
+                new_detection['DETECTION'] = detections_list
+                del new_detection['DETECTION_LIST']
+                fetched_events.append(new_detection)
+        else:
+            del host['DETECTION_LIST']
+            host['DETECTION'] = {}
+            fetched_events.append(host)
+    return fetched_events
+
+
+def get_activity_logs_events(client, since_datetime, max_fetch, next_page=None) -> tuple[Optional[list], dict]:
+    """ Get logs activity from qualys
```

Comment:
let's add a bit more details here about how we parse the response and the next page and how we handle the dedup in the last page.

Reply:
I explained it in the fetch function with comments

Reply:
Added.